src/attachmentlistmodel.cpp

```diff
@@ -150,17 +149,12 @@ void AttachmentListModel::onAttachmentPathChanged(const QString &attachmentLocat
 {
     for (int i = 0; i < m_attachmentsList.count(); ++i) {
         Attachment *attachment = m_attachmentsList.at(i);
-        if (attachment->location == attachmentLocation) {
-            // Make url more secure and compatible with QMF client
-            QString downloads = downloadFolder(m_message, attachmentLocation);
-            QString fileName = path;
-            QString secureUrl = QUrl::fromLocalFile(downloads + "/" + fileName.remove(downloads).remove('/')).toString();
-            if (attachment->url != secureUrl) {
-                attachment->url = secureUrl;
+        if (attachment->location == attachmentLocation
+            && attachment->url != path) {
+                attachment->url = path;
```

Comment:
Looks suspicious. A path is not a url and there should be attention on separating the two not to get bitten by url escaping rules. Moreover there was some old bug here with specifically crafted emails on referring to any file and wondering if the old problem is being reintroduced.

Reply:
Very good point. I was a bit hasty removing the QUrl(path).toString() part. I must stay there. I'm still removing the "securing" part which consist in removing the '/' characters from the name. When looking at what is emitting the attachmentPathChanged signal, only the EmailAgent matches in two places. The PR is centralising the changes there. With the first commit of the PR, the agent is guaranteed to send a valid path, with the download path prepend to a filename without any '/' character.

I've amended the commit to restore the Qurl().toString() part.